Add 1.55.0 post (#890)
rust-lang/blog.rust-lang.org: posts/2021-09-09-Rust-1.55.0.md

    Checking foo v0.1.0
warning: function is never used: `foo`
 --> src/lib.rs:9:4
  |
9 | fn foo() {}
  |    ^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: 1 warning emitted

warning: function is never used: `foo`
 --> src/lib.rs:9:4
  |
9 | fn foo() {}
  |    ^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: 1 warning emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
```

In 1.55, this behavior has been adjusted to deduplicate and print a report at the end of compilation:

```
$ cargo +1.55.0 check --all-targets
    Checking foo v0.1.0
warning: function is never used: `foo`
 --> src/lib.rs:9:4
  |
9 | fn foo() {}
  |    ^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: `foo` (lib) generated 1 warning
warning: `foo` (lib test) generated 1 warning (1 duplicate)
    Finished dev [unoptimized + debuginfo] target(s) in 0.84s
```

### Faster, more correct float parsing

The standard library's implementation of float parsing has been updated to use the Eisel-Lemire algorithm, which brings both speed improvements and improved correctness. In the past, certain edge cases failed to parse, and this has now been fixed.

> Read more details on the new implementation [here](https://github.com/rust-lang/rust/pull/86761).

Reviewer: Re-worded it slightly to be more of a call to action, and replaced "here" with a description for the link, as "here" is generally considered bad style for links.
```suggestion
You can read more details on the new implementation [in the pull request description.](https://github.com/rust-lang/rust/pull/86761).
```

> You can read more details on the new implementation [in the pull request description.](https://github.com/rust-lang/rust/pull/86761).

Reviewer 2: There's already a dot after the link 

```suggestion
You can read more details on the new implementation [in the pull request description](https://github.com/rust-lang/rust/pull/86761).
```

Rust code should never match `ErrorKind::Other` and expect any particular underlying error code; only match `ErrorKind::Other` if you're catching a constructed `std::io::Error` that uses that error kind. Rust code matching on `std::io::Error` should always use `_` for any error kinds it doesn't know about, in which case it can match the underlying error code, or report the error, or bubble it up to calling code.

We're making this change to smooth the way for introducing new ErrorKind variants in the future; those new variants will start out nightly-only, and only become stable later. This change ensures that code matching variants it doesn't know about must use a catch-all `_` pattern, which will work both with `ErrorKind::Uncategorized` and with future nightly-only variants.

[`std::io::ErrorKind`]: https://doc.rust-lang.org/stable/std/io/enum.ErrorKind.html

### Open range patterns added

Rust 1.55 stabilized using open ranges in patterns:

> ```rust=

Reviewer 2: ```suggestion
```rust
```

- [`x86_64::_bittestandreset64`]
- [`x86_64::_bittestandset64`]

The following previously stable functions are now `const`.

- [`str::from_utf8_unchecked`]

[`array::map`]: https://doc.rust-lang.org/stable/std/primitive.array.html#method.map
[`Bound::cloned`]: https://doc.rust-lang.org/stable/std/ops/enum.Bound.html#method.cloned
[`Drain::as_str`]: https://doc.rust-lang.org/stable/std/string/struct.Drain.html#method.as_str
[`IntoInnerError::into_error`]: https://doc.rust-lang.org/stable/std/io/struct.IntoInnerError.html#method.into_error
[`IntoInnerError::into_parts`]: https://doc.rust-lang.org/stable/std/io/struct.IntoInnerError.html#method.into_parts
[`MaybeUninit::assume_init_mut`]: https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.assume_init_mut
[`MaybeUninit::assume_init_ref`]: https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.assume_init_ref
[`MaybeUninit::write`]: https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.write
[`Seek::rewind`]: https://doc.rust-lang.org/stable/std/io/trait.Seek.html#method.rewind
[`ops::ControlFlow`]: https://doc.rust-lang.org/stable/std/ops/enum.ControlFlow.html

> [`str::from_utf8_unchecked`]: https://doc.rust-lang.org/stable/std/str/fn.from_utf8_unchecked.html

Reviewer 3: This has always been stable?!

Author: It's now usable in const contexts.

Reviewer 3: Oh, sorry misunderstood that section.
